kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/web/struts/RequisitionAction.java

```
@@ -0,0 +1,108 @@
+     * do not have as restricted delivery options.
+     */
+    protected void updateRouteCode(RequisitionDocument document) {
+        String routeCodeValue = null;
+        if ( StringUtils.equals(PurapConstants.RequisitionSources.B2B, document.getRequisitionSourceCode())) {
+            BusinessObjectService bos = SpringContext.getBean(BusinessObjectService.class);
+            
+            if(StringUtils.isNotBlank(document.getDeliveryBuildingCode())) {
+                Map<String, String> primaryKeys = new HashMap<String, String>();
+                primaryKeys.put(KFSPropertyConstants.BUILDING_CODE, document.getDeliveryBuildingCode());
+                List<Building> buildings = (List<Building>) bos.findMatching(Building.class, primaryKeys);
+                Building building =  (Building) buildings.get(0);
+                BuildingExtension buildingExt = (BuildingExtension) building.getExtension();
+                routeCodeValue = buildingExt.getRouteCode();
+            }
+            
+            document.setRouteCode(routeCodeValue);
+
+            if (StringUtils.isBlank(routeCodeValue)) {
+                document.setAddressToVendorIndicator(false); //default to final delivery address
+            }else{
+                try {
+                    ParameterEvaluator param;
+                    param = getParameterEvaluatorService().getParameterEvaluator(PurchaseOrderDocument.class, PurapConstants.B2B_DIRECT_SHIP_ROUTE_CODES_PARM, routeCodeValue.toUpperCase());
+
+                    if (param.getValue().contains(routeCodeValue)) {
+                        document.setAddressToVendorIndicator(true); //default to receiving address
+                    } else {
+                        document.setAddressToVendorIndicator(false); //default to final delivery address
+                    }
+                } catch ( Exception ex ) {
+                    LOG.error( "Exception while determining how to default delivery address.", ex );
```

The error is logged but the exception is swallowed. Please add a comment if this is acceptable and intended and why. I'm thinking that it is acceptable since we are simply defaulting an address here. Otherwise, rethrow.